pkg/lib/authn/authenticator/password/provider.go

```diff
@@ -93,14 +94,15 @@ func (p *Provider) WithSpec(a *authenticator.Password, spec *authenticator.Passw
 		return nil, err
 	}
 
-	// If password is not changed, skip the logic.
-	// Return original authenticator pointer
-	if pwd.Compare([]byte(password), a.PasswordHash) == nil {
-		return a, nil
+	newAuthen := a
+	if pwd.Compare([]byte(password), a.PasswordHash) != nil {
+		newAuthen = p.populatePasswordHash(a, password)
+	}
+
+	if !spec.ExpireAfter.Equal(*a.ExpireAfter) {
```

Comment:
spec.ExpireAfter can be nil

Reply:
Once we take UpdatePasswordOptions here, we do not need to compare ExpireAfter. We just look at SetExpireAfter, if it is true, then we set a.ExpireAfter to UpdatePasswordOptions.ExpireAfter. If it is false, we do not set.